colossus-metrics/src/main/scala/colossus/metrics/Collection.scala

```diff
@@ -97,18 +97,22 @@ private[metrics] class CollectionMap[T] {
 
 class DuplicateMetricException(message: String) extends Exception(message)
 
-class Collection(val config: CollectorConfig) {
-
-  val collectors : ConcurrentHashMap[MetricAddress, Collector] = new ConcurrentHashMap[MetricAddress, Collector]()
+class Collection(val config: CollectorConfig)
```

Comment:
This class definitely needs some docs now explaining what it is, how it works, where it's used, and also why we've chosen to use a trie structure.